logbook-servlet/src/main/java/org/zalando/logbook/servlet/RemoteRequest.java

```diff
@@ -82,22 +84,39 @@ public String getQuery() {
         return builder.build();
     }
 
+    @SuppressWarnings("unchecked") // warnings appear when using Servlet API 2.5
+    @Override
+    public Map<String, String[]> getParameterMap() {
+        Map<String, String[]> parameterMap = super.getParameterMap();
+        if (getContentType() == null || !getContentType().toLowerCase().startsWith("application/x-www-form-urlencoded")) {
+            return parameterMap;
+        }
+        String formEncoded = parameterMap.entrySet().stream()
+                .flatMap(entry -> stream(entry.getValue()).map(value -> entry.getKey() + "=" + value))
+                .collect(joining("&"));
+        body = formEncoded.getBytes(getCharset());
```

Comment:
This is a rather nasty side-effect. Can't we move this decision somewhere else? Technically this would need to happen for all `getParameter*` methods, not just `getParameterMap`.